pkg/healthcheck/healthcheck.go

```diff
@@ -1428,6 +1427,11 @@ func (hc *HealthChecker) allCategories() []*Category {
 					description: "opaque ports are properly annotated",
 					hintAnchor:  "linkerd-opaque-ports-definition",
 					check: func(ctx context.Context) error {
+						if hc.DataPlaneNamespace == "" {
+							// when checking proxies in all namespaces, this check is a no-op
+							return nil
+						}
```

Comment:
So right now when running `linkerd check` without specifying a namespace, the check does not issue any warnings anymore.

I think when a namespace is not specified, we should default to all namespaces. That way, when using the example I provided above when I install the emojivoto app and annotate a voting pod but not the voting service, we still see an error issued.

If we remove this check right here, we can actually remove the `namespace` argument from the `checkMisconfiguredOpaquePortAnnotations` function.

This is because in `checkMisconfiguredOpaquePortAnnotations` when we get services with `services, err := hc.GetServices(ctx)`, we have already use the `hc.DataPlaneNamespace` if it is specified. If it is not set, then we default to all namespaces.

So, if we remove this check and remove the `namespace` argument from `checkMisconfiguredOpaquePortAnnotations`, this works as I would expect:

```
$ linkerd check
...
linkerd-opaque-ports-definition
-------------------------------
× opaque ports are properly annotated
    pod/voting-ff4c54b8d-h8pdc has the annotation but service/voting-svc doesn't
    see https://linkerd.io/2/checks/#linkerd-opaque-ports-definition for hints

Status check results are ×

$ linkerd check --emojivoto
...
linkerd-opaque-ports-definition
-------------------------------
× opaque ports are properly annotated
    pod/voting-ff4c54b8d-h8pdc has the annotation but service/voting-svc doesn't
    see https://linkerd.io/2/checks/#linkerd-opaque-ports-definition for hints

Status check results are ×

$ linkerd check --linkerd
...
linkerd-opaque-ports-definition
-------------------------------
√ opaque ports are properly annotated

Status check results are √
```

Reply:
Hello @kleimkuhler ,

I think this has been my bad, sorry. I introduced this because getting a concrete resource by name requires a valid namespace to be passed, otherwise you get an error similar to

```
an empty namespace may not be set when a resource name is provided
```

when you perform an operation like:

```go
endpoint, err := hc.kubeAPI.CoreV1().Endpoints(hc.DataPlaneNamespace).Get(ctx, service.Name, metav1.GetOptions{})
```

I guess we could perform a list operation with a `LabelSelector` and perform a lookup locally against all the results matching the query. I think the same applies while retrieving a particular pod:

```go
pod, err := hc.kubeAPI.CoreV1().Pods(hc.DataPlaneNamespace).Get(ctx, addr.TargetRef.Name, metav1.GetOptions{})
```

I am not intimately familiar with the Go Kube SDK and, of course, I could be missing something obvious :D

Reply:
@migue all good! Yep so when getting the `endpoint` and `pod` you would just pass the `service.Namespace`. It's similar to how we use `service.Name` for the actual service name.